state/factory/workingset.go

```diff
@@ -225,30 +249,32 @@ func (ws *workingSet) Finalize() error {
 		return errors.New("Cannot finalize a working set twice")
 	}
 	ws.finalized = true
-	// Persist accountTrie's root hash
-	rootHash := ws.accountTrie.RootHash()
-	ws.cb.Put(AccountKVNameSpace, []byte(AccountTrieRootKey), rootHash, "failed to store accountTrie's root hash")
 	// Persist current chain Height
 	h := byteutil.Uint64ToBytes(ws.blockHeight)
-	ws.cb.Put(AccountKVNameSpace, []byte(CurrentHeightKey), h, "failed to store accountTrie's current Height")
+	if err := ws.dao.Put(AccountKVNamespace, []byte(CurrentHeightKey), h); err != nil {
+		log.S().Panic("Unexpected error", zap.Error(err))
```

Comment:
"failed to write factory height"?

Reply:
that works too. The root reason of using it in this way is that kvstorebuffer.Put suppose to always return nil. I would wrap it prevent the panic call.